src/analyzers/collectors/sshCollector.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+from argparse import Namespace
+from queue import Queue
+import logging
+import random
+import stat
+from pathlib import Path
+import sys
+import time
+from typing import Dict, List
+
+import paramiko
+
+from src.analyzers.collectors.perfParser import PerfParser, TestRes
+from src.helpers.backGroundBuilder import CSignal
+
+
+class SshCollector:
+    def __init__(self, settings: Namespace, bin_dir: Path | None = None):
+        self.settings = settings
+        self.max_test_launches = settings.__dict__.get("max_test_launches", -1)
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(self.settings.log_level)
+        self.is_tmp_created = False
+
+        host = settings.__dict__.get("host", "127.0.0.1")
```

Comment:
Sould be rewrited with vars

Reply:
Maybe create separate PR with commit which fix all `__dict__` -> `vars()`

Reply:
Good idea